From 2264a62d4af18e52d058c89a0c656e0b6c775459 Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Sun, 19 Jan 2020 11:29:21 +0800
Subject: [PATCH] malta64: Remove __init annotation of
 malta_pcnet32_ioports_fixup

commit 445b8807289d8647c5fcf035ca36f52a9a942a75 linux-yocto upstream

To fix the following warning.

WARNING: vmlinux.o(.pci_fixup_final+0x1288): Section mismatch in reference
from the variable __pci_fixup_malta_pcnet32_ioports_fixup163 to the
function .init.text:malta_pcnet32_ioports_fixup()
The variable __pci_fixup_malta_pcnet32_ioports_fixup163 references
a function __init malta_pcnet32_ioports_fixup().
This is often seen when error handling in the init function
uses functionality in the exit path.
The fix is often to remove the __init annotation of
malta_pcnet32_ioports_fixup() so it may be used outside an exit section.

Fixes: b73edd2e3a58 ("mti_malta32: fix the pci resource conflicts")
Signed-off-by: He Zhe <zhe.he@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/mips/pci/fixup-malta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/pci/fixup-malta.c b/arch/mips/pci/fixup-malta.c
index ffc068e7f13e..df1cae42ff9e 100644
--- a/arch/mips/pci/fixup-malta.c
+++ b/arch/mips/pci/fixup-malta.c
@@ -143,7 +143,7 @@ DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_INTEL, PCI_DEVICE_ID_INTEL_82371AB,
 	 malta_piix_func1_fixup);
 
 /* Use the io resource allocated by YAMON */
-static void __init malta_pcnet32_ioports_fixup(struct pci_dev *pdev)
+static void malta_pcnet32_ioports_fixup(struct pci_dev *pdev)
 {
 	u32 check = 0, new = 0x1060;
 
-- 
2.26.1

